include/librealsense2/dds/dds-participant.cpp

```diff
@@ -84,6 +85,18 @@ struct dds_participant::listener_impl
             break;
         }
     }
+
+    virtual void on_type_discovery( eprosima::fastdds::dds::DomainParticipant * participant,
+                                    const eprosima::fastrtps::rtps::SampleIdentity & request_sample_id,
+                                    const eprosima::fastrtps::string_255 & topic_name,
+                                    const eprosima::fastrtps::types::TypeIdentifier * identifier,
+                                    const eprosima::fastrtps::types::TypeObject * object,
+                                    eprosima::fastrtps::types::DynamicType_ptr dyn_type ) override
+    {
+        TypeSupport type_support( new eprosima::fastrtps::types::DynamicPubSubType( dyn_type ) );
```

Comment:
Seem weird to do this allocation just to get the name...

Reply:
I added it since you added debug logs for the other callbacks, and it gives us details of the underlying structure type. It happens only during discovery phase, once for each dynamic topic discovered, so I don't think will impact performance.